fly/main.go

```diff
@@ -49,12 +49,19 @@ func handleError(helpParser *flags.Parser, err error) {
 			fmt.Fprintln(ui.Stderr, "")
 			fmt.Fprintln(ui.Stderr, "    "+ui.Embolden("fly -t %s login", commands.Fly.Target))
 			fmt.Fprintln(ui.Stderr, "")
-		} else if err == rc.ErrNoTargetSpecified {
-			fmt.Fprintln(ui.Stderr, "no target specified. specify the target with "+ui.Embolden("-t")+" or log in like so:")
+		} else if err == rc.ErrNoTargetSpecifiedNoneAvailable {
```

Comment:
It might be worth adding tests for this behavior, since otherwise this isn't covered - the tests you added cover the conditions under which the error is returned, but not the messaging to the user.

It looks like some tests are [failing](https://ci.concourse-ci.org/builds/5169459#L5c98f076:519:527) due to the new message so maybe that suite could just be updated?